src/proc_diskstats.c

```diff
@@ -523,6 +678,61 @@ int do_proc_diskstats(int update_every, unsigned long long dt) {
 		}
 
 		// --------------------------------------------------------------------
+
+		if(ddo_space) {
+			if(mount_point) {
+				if (statvfs(family, &buff_statvfs) < 0) {
+					error("Faild checking disk space usage of %s", family);
+				} else {
+					space_avail = buff_statvfs.f_bavail * buff_statvfs.f_bsize;
+					space_avail_root = (buff_statvfs.f_bfree - buff_statvfs.f_bavail) * buff_statvfs.f_bsize;
+					space_used = (buff_statvfs.f_blocks - buff_statvfs.f_bfree) * buff_statvfs.f_bsize;
+				}
+
+				st = rrdset_find_bytype("disk_space", disk);
```

Comment:
Something is wrong here.
If `statvfs()` fails, it will add the chart / refresh the values, while it should not.

Also, if all values read by `statvfs()` are zero and the `ddo_space == CONFIG_ONDEMAND_ONDEMAND` charts should not be added either.